classes/plugins/PKPPubIdPlugin.inc.php

```diff
@@ -0,0 +1,428 @@
+				$contextDao->deleteAllPubIds($context->getId(), $this->getPubIdType());
+				$notificationManager->createTrivialNotification($user->getId(), NOTIFICATION_TYPE_SUCCESS);
+				return new JSONMessage(true);
+			default:
+				$form->initData();
+				return new JSONMessage(true, $form->fetch($request));
+		}
+		return parent::manage($args, $request);
+	}
+
+
+	//
+	// Protected template methods to be implemented by sub-classes.
+	//
+	/**
+	 * Get the public identifier.
+	 * @param $pubObject object
+	 * 	OJS Issue, Article, Representation or SubmissionFile
+	 * @return string
+	 */
+	abstract function getPubId($pubObject);
+
+	/**
+	 * Construct the public identifier from its prefix and suffix.
+	 * @param $pubIdPrefix string
+	 * @param $pubIdSuffix string
+	 * @param $contextId integer
+	 * @return string
+	 */
+	abstract function constructPubId($pubIdPrefix, $pubIdSuffix, $contextId);
+
+	/**
+	 * Public identifier type, see
+	 * http://dtd.nlm.nih.gov/publishing/tag-library/n-4zh0.html
+	 * @return string
+	 */
+	abstract function getPubIdType();
+
+	/**
+	 * Public identifier type that will be displayed to the reader.
+	 * @return string
+	 */
+	abstract function getPubIdDisplayType();
+
+	/**
+	 * Full name of the public identifier.
+	 * @return string
+	 */
+	abstract function getPubIdFullName();
+
+	/**
+	 * Get the whole resolving URL.
+	 * @param $contextId integer
+	 * @param $pubId string
+	 * @return string resolving URL
+	 */
+	abstract function getResolvingURL($contextId, $pubId);
+
+	/**
+	 * Get the file (path + filename)
+	 * to be included into the object's
+	 * identifiers tab.
+	 * @return string
+	 */
+	abstract function getPubIdMetadataFile();
+
+	/**
+	 * Get the file (path + filename)
+	 * for the pub id assignment
+	 * to be included into other pages.
+	 * @return string
+	 */
+	abstract function getPubIdAssignFile();
+
+	/**
+	 * Get the settings form.
+	 * @param $contextId integer
+	 * @return object Settings form
+	 */
+	abstract function instantiateSettingsForm($contextId);
+
+	/**
+	 * Get the additional form field names.
+	 * @return array
+	 */
+	abstract function getFormFieldNames();
+
+	/**
+	 * Get the assign option form field name.
+	 * @return string
+	 */
+	abstract function getAssignFormFieldName();
+
+	/**
+	 * Get the the prefix form field name.
+	 * @return string
+	 */
+	abstract function getPrefixFieldName();
+
+	/**
+	 * Get the the suffix form field name.
+	 * @return string
+	 */
+	abstract function getSuffixFieldName();
+
+	/**
+	 * Get the link actions used in the pub id forms.
+	 * @return array
+	 */
+	abstract function getLinkActions($pubObject);
+
+	/**
+	 * Get the suffix patterns form field names for all objects.
+	 * @return array (pub object type => suffix pattern field name)
+	 */
+	abstract function getSuffixPatternsFieldNames();
+
+	/**
+	 * Get additional field names to be considered for storage.
+	 * @return array
+	 */
+	abstract function getDAOFieldNames();
```

Comment:
In some cases (like this one) I think it makes sense to provide a default implementation. In this case it would return an empty array. That way subclasses that don't make use of a certain feature of the parent class wouldn't need to implement this.